From 16276445b454ae8e206ccbc61f5e6a742c2a58ef Mon Sep 17 00:00:00 2001
From: UN1CA Build <un1ca@example.com>
Date: Sun, 26 Jul 2026 09:00:22 +0000
Subject: [PATCH] Add back right cutout support

Adapted for the current beyond0lte SystemUI.apk baseline: this build's
IndicatorGardenAlgorithm already contains the 3-arg
getLeftContainerMaxWidth/getRightContainerMaxWidth overloads upstream
(register-count smali, not locals-count), unlike the baseline the
original patch was written against, so the hunks are regenerated
against the real current source instead of force-applied.

Change-Id: Ib98ccd4e8ac8844953f4c91cf82e2f8eb96f720c
Signed-off-by: ExtremeXT <75576145+ExtremeXT@users.noreply.github.com>
---
 .../phone/IndicatorGardenAlgorithm.smali      | 187 ++++++++---
 .../IndicatorGardenAlgorithmFactory.smali     |  23 +-
 .../IndicatorGardenAlgorithmRightCutout.smali | 312 ++++++++++++++++++
 3 files changed, 469 insertions(+), 53 deletions(-)
 create mode 100644 smali_classes3/com/android/systemui/statusbar/phone/IndicatorGardenAlgorithmRightCutout.smali

diff --git a/smali_classes3/com/android/systemui/statusbar/phone/IndicatorGardenAlgorithm.smali b/smali_classes3/com/android/systemui/statusbar/phone/IndicatorGardenAlgorithm.smali
index 223cfed..f05e9eb 100644
--- a/smali_classes3/com/android/systemui/statusbar/phone/IndicatorGardenAlgorithm.smali
+++ b/smali_classes3/com/android/systemui/statusbar/phone/IndicatorGardenAlgorithm.smali
@@ -209,36 +209,58 @@
     return p0
 .end method
 
+.method public final getLeftContainerMaxWidth(Lcom/android/systemui/statusbar/phone/IndicatorGarden;)I
+    .registers 4
+
+    invoke-virtual {p0}, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithm;->getDefaultSidePadding()I
+
+    move-result v0
+
+    invoke-virtual {p0}, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithm;->getDefaultSidePadding()I
+
+    move-result v1
+
+    invoke-virtual {p0, p1, v0, v1}, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithm;->getLeftContainerMaxWidth(Lcom/android/systemui/statusbar/phone/IndicatorGarden;II)I
+
+    move-result p0
+
+    return p0
+.end method
+
 .method public final getLeftContainerMaxWidth(Lcom/android/systemui/statusbar/phone/IndicatorGarden;II)I
-    .registers 6
+    .registers 8
+
+    move v0, p2
+
+    move v1, p3
 
     invoke-interface {p1}, Lcom/android/systemui/statusbar/phone/IndicatorGarden;->getCenterContainer()Lcom/android/systemui/statusbar/phone/IndicatorGardenContainer;
 
-    move-result-object v0
+    move-result-object v2
 
     iget-object p0, p0, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithm;->inputProperties:Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;
 
-    if-eqz v0, :cond_21
+    if-eqz v2, :cond_23
 
-    invoke-interface {v0}, Lcom/android/systemui/statusbar/phone/IndicatorGardenContainer;->isGardenVisible()Z
+    invoke-interface {v2}, Lcom/android/systemui/statusbar/phone/IndicatorGardenContainer;->isGardenVisible()Z
 
-    move-result v1
+    move-result v3
 
-    if-eqz v1, :cond_21
+    if-eqz v3, :cond_23
 
-    invoke-interface {v0}, Lcom/android/systemui/statusbar/phone/IndicatorGardenContainer;->getGardenWidth()I
+    invoke-interface {v2}, Lcom/android/systemui/statusbar/phone/IndicatorGardenContainer;->getGardenWidth()I
 
-    move-result v1
+    move-result v3
 
-    if-lez v1, :cond_21
+    if-lez v3, :cond_23
 
     iget p0, p0, Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;->statusBarWidth:I
 
     div-int/lit8 p0, p0, 0x2
 
-    sub-int/2addr p0, p2
+    sub-int/2addr p0, v0
 
-    invoke-interface {v0}, Lcom/android/systemui/statusbar/phone/IndicatorGardenContainer;->getGardenWidth()I
+    invoke-interface {v2}, Lcom/android/systemui/statusbar/phone/IndicatorGardenContainer;->getGardenWidth()I
 
     move-result p1
 
@@ -248,48 +270,66 @@
 
     return p0
 
-    :cond_21
+    :cond_23
     invoke-interface {p1}, Lcom/android/systemui/statusbar/phone/IndicatorGarden;->getRightContainer()Lcom/android/systemui/statusbar/phone/IndicatorGardenContainer;
 
-    move-result-object v0
+    move-result-object v2
 
-    if-eqz v0, :cond_2c
+    if-eqz v2, :cond_2e
 
-    invoke-interface {v0}, Lcom/android/systemui/statusbar/phone/IndicatorGardenContainer;->getGardenWidth()I
+    invoke-interface {v2}, Lcom/android/systemui/statusbar/phone/IndicatorGardenContainer;->getGardenWidth()I
 
-    move-result v0
+    move-result v2
 
-    goto :goto_2d
+    goto :goto_2f
 
-    :cond_2c
-    const/4 v0, 0x0
+    :cond_2e
+    const/4 v2, 0x0
 
-    :goto_2d
+    :goto_2f
     invoke-interface {p1}, Lcom/android/systemui/statusbar/phone/IndicatorGarden;->getEssentialRightWidth()I
 
-    move-result v1
+    move-result v3
 
-    invoke-static {v1, v0}, Ljava/lang/Math;->max(II)I
+    invoke-static {v3, v2}, Ljava/lang/Math;->max(II)I
 
-    move-result v0
+    move-result v2
 
-    add-int/2addr p2, p3
+    add-int/2addr v0, v1
 
-    iget p3, p0, Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;->statusBarWidth:I
+    iget v1, p0, Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;->statusBarWidth:I
 
-    sub-int/2addr p3, v0
+    sub-int/2addr v1, v2
 
-    sub-int/2addr p3, p2
+    sub-int/2addr v1, v0
 
     iget p0, p0, Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;->defaultCenterPadding:I
 
-    sub-int/2addr p3, p0
+    sub-int/2addr v1, p0
 
     invoke-interface {p1}, Lcom/android/systemui/statusbar/phone/IndicatorGarden;->getEssentialLeftWidth()I
 
     move-result p0
 
-    invoke-static {p3, p0}, Ljava/lang/Math;->max(II)I
+    invoke-static {v1, p0}, Ljava/lang/Math;->max(II)I
+
+    move-result p0
+
+    return p0
+.end method
+
+.method public final getRightContainerMaxWidth(Lcom/android/systemui/statusbar/phone/IndicatorGarden;)I
+    .registers 4
+
+    invoke-virtual {p0}, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithm;->getDefaultSidePadding()I
+
+    move-result v0
+
+    invoke-virtual {p0}, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithm;->getDefaultSidePadding()I
+
+    move-result v1
+
+    invoke-virtual {p0, p1, v0, v1}, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithm;->getRightContainerMaxWidth(Lcom/android/systemui/statusbar/phone/IndicatorGarden;II)I
 
     move-result p0
 
@@ -297,35 +337,39 @@
 .end method
 
 .method public final getRightContainerMaxWidth(Lcom/android/systemui/statusbar/phone/IndicatorGarden;II)I
-    .registers 6
+    .registers 8
+
+    move v0, p2
+
+    move v1, p3
 
     invoke-interface {p1}, Lcom/android/systemui/statusbar/phone/IndicatorGarden;->getCenterContainer()Lcom/android/systemui/statusbar/phone/IndicatorGardenContainer;
 
-    move-result-object v0
+    move-result-object v2
 
     iget-object p0, p0, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithm;->inputProperties:Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;
 
-    if-eqz v0, :cond_21
+    if-eqz v2, :cond_23
 
-    invoke-interface {v0}, Lcom/android/systemui/statusbar/phone/IndicatorGardenContainer;->isGardenVisible()Z
+    invoke-interface {v2}, Lcom/android/systemui/statusbar/phone/IndicatorGardenContainer;->isGardenVisible()Z
 
-    move-result v1
+    move-result v3
 
-    if-eqz v1, :cond_21
+    if-eqz v3, :cond_23
 
-    invoke-interface {v0}, Lcom/android/systemui/statusbar/phone/IndicatorGardenContainer;->getGardenWidth()I
+    invoke-interface {v2}, Lcom/android/systemui/statusbar/phone/IndicatorGardenContainer;->getGardenWidth()I
 
-    move-result v1
+    move-result v3
 
-    if-lez v1, :cond_21
+    if-lez v3, :cond_23
 
     iget p0, p0, Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;->statusBarWidth:I
 
     div-int/lit8 p0, p0, 0x2
 
-    sub-int/2addr p0, p3
+    sub-int/2addr p0, v1
 
-    invoke-interface {v0}, Lcom/android/systemui/statusbar/phone/IndicatorGardenContainer;->getGardenWidth()I
+    invoke-interface {v2}, Lcom/android/systemui/statusbar/phone/IndicatorGardenContainer;->getGardenWidth()I
 
     move-result p1
 
@@ -335,32 +379,79 @@
 
     return p0
 
-    :cond_21
-    iget v0, p0, Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;->statusBarWidth:I
+    :cond_23
+    iget v2, p0, Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;->statusBarWidth:I
 
-    add-int/2addr p2, p3
+    add-int/2addr v0, v1
 
-    sub-int/2addr v0, p2
+    sub-int/2addr v2, v0
 
     invoke-interface {p1}, Lcom/android/systemui/statusbar/phone/IndicatorGarden;->getEssentialLeftWidth()I
 
-    move-result p2
+    move-result v0
 
-    sub-int/2addr v0, p2
+    sub-int/2addr v2, v0
 
     iget p0, p0, Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;->defaultCenterPadding:I
 
-    sub-int/2addr v0, p0
+    sub-int/2addr v2, p0
 
     invoke-interface {p1}, Lcom/android/systemui/statusbar/phone/IndicatorGarden;->getEssentialRightWidth()I
 
     move-result p0
 
-    invoke-static {v0, p0}, Ljava/lang/Math;->max(II)I
+    invoke-static {v2, p0}, Ljava/lang/Math;->max(II)I
+
+    move-result p0
+
+    return p0
+.end method
+
+.method public final getSideCutoutPadding(II)I
+    .registers 5
+
+    invoke-virtual {p0}, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithm;->getDefaultSidePadding()I
+
+    move-result v0
+
+    iget-object p0, p0, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithm;->inputProperties:Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;
+
+    iget v1, p0, Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;->rotation:I
+
+    if-nez v1, :cond_20
+
+    if-lez p1, :cond_19
+
+    iget p0, p0, Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;->cutoutSidePaddingD:I
+
+    add-int/2addr p1, p0
+
+    add-int/2addr p2, p0
+
+    invoke-static {p1, p2}, Ljava/lang/Math;->max(II)I
+
+    move-result p0
+
+    invoke-static {p0, v0}, Ljava/lang/Math;->max(II)I
+
+    move-result p0
+
+    goto :goto_1f
+
+    :cond_19
+    iget p0, p0, Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;->cornerPaddingC:I
+
+    invoke-static {p0, v0}, Ljava/lang/Math;->max(II)I
 
     move-result p0
 
+    :goto_1f
     return p0
+
+    :cond_20
+    add-int/2addr v0, p2
+
+    return v0
 .end method
 
 .method public hasCameraBottomMargin()Z
diff --git a/smali_classes3/com/android/systemui/statusbar/phone/IndicatorGardenAlgorithmFactory.smali b/smali_classes3/com/android/systemui/statusbar/phone/IndicatorGardenAlgorithmFactory.smali
index 42c39be..ec3e4e5 100644
--- a/smali_classes3/com/android/systemui/statusbar/phone/IndicatorGardenAlgorithmFactory.smali
+++ b/smali_classes3/com/android/systemui/statusbar/phone/IndicatorGardenAlgorithmFactory.smali
@@ -10,12 +10,14 @@
 
 .field public final indicatorGardenAlgorithmNoCutout:Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmNoCutout;
 
+.field public final indicatorGardenAlgorithmRightCutout:Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmRightCutout;
+
 .field public final indicatorGardenAlgorithmSidelingCenterCutout:Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmSidelingCenterCutout;
 
 
 # direct methods
-.method public constructor <init>(Lcom/android/systemui/statusbar/phone/IndicatorCutoutUtil;Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmNoCutout;Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmCenterCutout;Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmSidelingCenterCutout;)V
-    .registers 5
+.method public constructor <init>(Lcom/android/systemui/statusbar/phone/IndicatorCutoutUtil;Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmNoCutout;Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmCenterCutout;Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmSidelingCenterCutout;Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmRightCutout;)V
+    .registers 6
 
     invoke-direct {p0}, Ljava/lang/Object;-><init>()V
 
@@ -27,6 +29,8 @@
 
     iput-object p4, p0, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmFactory;->indicatorGardenAlgorithmSidelingCenterCutout:Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmSidelingCenterCutout;
 
+    iput-object p5, p0, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmFactory;->indicatorGardenAlgorithmRightCutout:Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmRightCutout;
+
     return-void
 .end method
 
@@ -57,19 +61,28 @@
     return-object p0
 
     :cond_12
+    sget-object v2, Lcom/android/systemui/statusbar/phone/CutoutType;->RIGHT_CUTOUT:Lcom/android/systemui/statusbar/phone/CutoutType;
+
+    if-ne v1, v2, :cond_19
+
+    iget-object p0, p0, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmFactory;->indicatorGardenAlgorithmRightCutout:Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmRightCutout;
+
+    return-object p0
+
+    :cond_19
     sget-boolean v1, Lcom/android/systemui/BasicRune;->STATUS_LAYOUT_SIDELING_CUTOUT:Z
 
-    if-eqz v1, :cond_1d
+    if-eqz v1, :cond_24
 
     invoke-virtual {v0}, Lcom/android/systemui/statusbar/phone/IndicatorCutoutUtil;->isMainDisplay()Z
 
     move-result v0
 
-    if-eqz v0, :cond_1d
+    if-eqz v0, :cond_24
 
     return-object v3
 
-    :cond_1d
+    :cond_24
     iget-object p0, p0, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmFactory;->indicatorGardenAlgorithmNoCutout:Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmNoCutout;
 
     return-object p0
diff --git a/smali_classes3/com/android/systemui/statusbar/phone/IndicatorGardenAlgorithmRightCutout.smali b/smali_classes3/com/android/systemui/statusbar/phone/IndicatorGardenAlgorithmRightCutout.smali
new file mode 100644
index 0000000..6cf6e4e
--- /dev/null
+++ b/smali_classes3/com/android/systemui/statusbar/phone/IndicatorGardenAlgorithmRightCutout.smali
@@ -0,0 +1,312 @@
+.class public final Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmRightCutout;
+.super Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithm;
+.source "Mesa & Renox, just like bare rocks"
+
+# instance fields
+.field public cutoutLeft:I
+
+.field public final inputProperties:Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;
+
+
+# direct methods
+.method public constructor <init>(Landroid/content/Context;Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;)V
+    .locals 0
+
+    invoke-direct {p0, p1, p2}, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithm;-><init>(Landroid/content/Context;Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;)V
+
+    iput-object p2, p0, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmRightCutout;->inputProperties:Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;
+
+    return-void
+.end method
+
+
+# virtual methods
+.method public final calculateCameraTopMargin()I
+    .locals 1
+
+    invoke-virtual {p0}, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmRightCutout;->hasCameraTopMargin()Z
+
+    move-result v0
+
+    if-eqz v0, :cond_0
+
+    iget-object p0, p0, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmRightCutout;->inputProperties:Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;
+
+    iget p0, p0, Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;->cutoutTopMarginB:I
+
+    goto :goto_0
+
+    :cond_0
+    const/4 p0, 0x0
+
+    :goto_0
+    return p0
+.end method
+
+.method public final calculateCenterContainerMaxWidth()I
+    .locals 1
+
+    invoke-virtual {p0}, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithm;->getDefaultSidePadding()I
+
+    move-result v0
+
+    mul-int/lit8 v0, v0, 0x2
+
+    iget-object p0, p0, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithm;->inputProperties:Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;
+
+    iget p0, p0, Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;->statusBarWidth:I
+
+    sub-int/2addr p0, v0
+
+    int-to-float p0, p0
+
+    const/high16 v0, 0x40400000    # 3.0f
+
+    div-float/2addr p0, v0
+
+    float-to-int p0, p0
+
+    const/4 v0, 0x0
+
+    invoke-static {p0, v0}, Ljava/lang/Math;->max(II)I
+
+    move-result p0
+
+    return p0
+.end method
+
+.method public final calculateLeftContainerMaxWidth(Lcom/android/systemui/statusbar/phone/IndicatorGarden;)I
+    .locals 3
+
+    iget-object v0, p0, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmRightCutout;->inputProperties:Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;
+
+    iget-object v0, v0, Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;->displayCutout:Landroid/view/DisplayCutout;
+
+    const/4 v1, 0x0
+
+    if-eqz v0, :cond_0
+
+    iget-object v0, p0, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmRightCutout;->inputProperties:Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;
+
+    iget-object v0, v0, Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;->displayCutout:Landroid/view/DisplayCutout;
+
+    invoke-static {v0}, Lkotlin/jvm/internal/Intrinsics;->checkNotNull(Ljava/lang/Object;)V
+
+    invoke-virtual {v0}, Landroid/view/DisplayCutout;->getSafeInsetLeft()I
+
+    move-result v1
+
+    iget-object v0, p0, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmRightCutout;->inputProperties:Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;
+
+    iget-object v0, v0, Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;->displayCutout:Landroid/view/DisplayCutout;
+
+    invoke-static {v0}, Lkotlin/jvm/internal/Intrinsics;->checkNotNull(Ljava/lang/Object;)V
+
+    invoke-virtual {v0}, Landroid/view/DisplayCutout;->getSafeInsetRight()I
+
+    move-result v0
+
+    goto :goto_0
+
+    :cond_0
+    move v0, v1
+
+    :goto_0
+    invoke-virtual {p0}, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmRightCutout;->calculateLeftPadding()I
+
+    move-result v2
+
+    sub-int/2addr v2, v1
+
+    invoke-virtual {p0}, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmRightCutout;->calculateRightPadding()I
+
+    move-result v1
+
+    sub-int/2addr v1, v0
+
+    invoke-virtual {p0, p1, v2, v1}, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithm;->getLeftContainerMaxWidth(Lcom/android/systemui/statusbar/phone/IndicatorGarden;II)I
+
+    move-result p0
+
+    return p0
+.end method
+
+.method public final calculateLeftPadding()I
+    .locals 3
+
+    iget-object v0, p0, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmRightCutout;->inputProperties:Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;
+
+    iget-object v1, v0, Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;->displayCutout:Landroid/view/DisplayCutout;
+
+    const/4 v2, 0x0
+
+    if-eqz v1, :cond_0
+
+    iget-object v0, v0, Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;->displayCutout:Landroid/view/DisplayCutout;
+
+    invoke-static {v0}, Lkotlin/jvm/internal/Intrinsics;->checkNotNull(Ljava/lang/Object;)V
+
+    invoke-virtual {v0}, Landroid/view/DisplayCutout;->getSafeInsetLeft()I
+
+    move-result v0
+
+    goto :goto_0
+
+    :cond_0
+    move v0, v2
+
+    :goto_0
+    invoke-virtual {p0, v2, v0}, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithm;->getSideCutoutPadding(II)I
+
+    move-result p0
+
+    return p0
+.end method
+
+.method public final calculateRightContainerMaxWidth(Lcom/android/systemui/statusbar/phone/IndicatorGarden;)I
+    .locals 3
+
+    iget-object v0, p0, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmRightCutout;->inputProperties:Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;
+
+    iget-object v0, v0, Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;->displayCutout:Landroid/view/DisplayCutout;
+
+    const/4 v1, 0x0
+
+    if-eqz v0, :cond_0
+
+    iget-object v0, p0, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmRightCutout;->inputProperties:Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;
+
+    iget-object v0, v0, Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;->displayCutout:Landroid/view/DisplayCutout;
+
+    invoke-static {v0}, Lkotlin/jvm/internal/Intrinsics;->checkNotNull(Ljava/lang/Object;)V
+
+    invoke-virtual {v0}, Landroid/view/DisplayCutout;->getSafeInsetLeft()I
+
+    move-result v1
+
+    iget-object v0, p0, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmRightCutout;->inputProperties:Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;
+
+    iget-object v0, v0, Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;->displayCutout:Landroid/view/DisplayCutout;
+
+    invoke-static {v0}, Lkotlin/jvm/internal/Intrinsics;->checkNotNull(Ljava/lang/Object;)V
+
+    invoke-virtual {v0}, Landroid/view/DisplayCutout;->getSafeInsetRight()I
+
+    move-result v0
+
+    goto :goto_0
+
+    :cond_0
+    move v0, v1
+
+    :goto_0
+    invoke-virtual {p0}, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmRightCutout;->calculateLeftPadding()I
+
+    move-result v2
+
+    sub-int/2addr v2, v1
+
+    invoke-virtual {p0}, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmRightCutout;->calculateRightPadding()I
+
+    move-result v1
+
+    sub-int/2addr v1, v0
+
+    invoke-virtual {p0, p1, v2, v1}, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithm;->getRightContainerMaxWidth(Lcom/android/systemui/statusbar/phone/IndicatorGarden;II)I
+
+    move-result p0
+
+    return p0
+.end method
+
+.method public final calculateRightPadding()I
+    .locals 3
+
+    iget-object v0, p0, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmRightCutout;->inputProperties:Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;
+
+    iget v1, v0, Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;->statusBarWidth:I
+
+    iget v2, p0, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmRightCutout;->cutoutLeft:I
+
+    sub-int/2addr v1, v2
+
+    iget-object v2, v0, Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;->displayCutout:Landroid/view/DisplayCutout;
+
+    if-eqz v2, :cond_0
+
+    iget-object v0, v0, Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;->displayCutout:Landroid/view/DisplayCutout;
+
+    invoke-static {v0}, Lkotlin/jvm/internal/Intrinsics;->checkNotNull(Ljava/lang/Object;)V
+
+    invoke-virtual {v0}, Landroid/view/DisplayCutout;->getSafeInsetRight()I
+
+    move-result v0
+
+    goto :goto_0
+
+    :cond_0
+    const/4 v0, 0x0
+
+    :goto_0
+    invoke-virtual {p0, v1, v0}, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithm;->getSideCutoutPadding(II)I
+
+    move-result p0
+
+    return p0
+.end method
+
+.method public final hasCameraTopMargin()Z
+    .locals 2
+
+    invoke-virtual {p0}, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithm;->getHasCutoutForIndicator()Z
+
+    move-result v0
+
+    const/4 v1, 0x0
+
+    if-eqz v0, :cond_1
+
+    iget-object p0, p0, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithm;->inputProperties:Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;
+
+    iget p0, p0, Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;->rotation:I
+
+    const/4 v0, 0x1
+
+    if-nez p0, :cond_0
+
+    move p0, v0
+
+    goto :goto_0
+
+    :cond_0
+    move p0, v1
+
+    :goto_0
+    if-eqz p0, :cond_1
+
+    move v1, v0
+
+    :cond_1
+    return v1
+.end method
+
+.method public final initResources()V
+    .locals 1
+
+    iget-object v0, p0, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmRightCutout;->inputProperties:Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;
+
+    iget-object v0, v0, Lcom/android/systemui/statusbar/phone/IndicatorGardenInputProperties;->displayCutout:Landroid/view/DisplayCutout;
+
+    if-eqz v0, :cond_0
+
+    invoke-virtual {v0}, Landroid/view/DisplayCutout;->getBoundingRectTop()Landroid/graphics/Rect;
+
+    move-result-object v0
+
+    iget v0, v0, Landroid/graphics/Rect;->left:I
+
+    iput v0, p0, Lcom/android/systemui/statusbar/phone/IndicatorGardenAlgorithmRightCutout;->cutoutLeft:I
+
+    :cond_0
+    return-void
+.end method
\ No newline at end of file
-- 
2.43.0
